### PR TITLE
Tests should use theory data

### LINKED ISSUE #11041: Tests should use TheoryData<>
Theory tests should preferably use the `TheoryData<>` pattern where possible. We have several hundred hits for the analyzer for this currently (`xUnit1042`). The analyzer is disabled in the `.editorconfig` file and can be commented out to hunt these down.

Once addressed we can turn this back on.

### PR DESCRIPTION
**Pull request by LeafShi1**
Makes progress to fix #11041 
Deal with the feedback of the PR #11171 

## Proposed changes

- Changed untyped test data to use typed TheoryData objects.
- Allowing us to enable xUnit1024 errors in .editorconfig.
- Currently fixes test cases in Windows form analyzer (CSharp) with one exception 
   for the file reader font converter.

## Customer Impact
- No customer impact, just refactoring old test cases.

## Regression? 

- No

## Risk

-There should be no risk in moving to strongly typed parameterized test methods.

## Test methodology 
- Because the tests are functionally identical, I used the tests that were already present. 

 

## Test environment(s) 

- 

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11171)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -646,9 +646,6 @@ dotnet_diagnostic.CA1861.severity = silent
 # This collides with TheoryData usage
 dotnet_style_prefer_collection_expression = true:silent
 
-# This should be fixed https://github.com/dotnet/winforms/issues/11041
```
Comment (Tanya-Solyanik) [inline .editorconfig:649]: Does this PR resolve ass cases of untyped TheoryData? I see that the CI build passed, but I wonder if you are getting build warning on command line or in the VS?

File: src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/System/Windows/Forms/Generators/ProjectFileReaderTests.FontConverter.cs [hunk 2]
```diff
@@ -57,7 +57,9 @@ public static IEnumerable<object[]> TestConvertFormData()
         }
 
         [Theory]
+#pragma warning disable xUnit1042 // The member referenced by the MemberData attribute returns untyped data rows
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/System/Windows/Forms/Generators/ProjectFileReaderTests.FontConverter.cs:60]: Please fix this one as well.

File: .editorconfig (hunk 1 above)
Comment (LeafShi1) [reply .editorconfig:649]: No, I will add this back

image

But if keeping the code here, the `#pragma warning disable xUnit1042` added in file ProjectFileReaderTests.FontConverter.cs will generate error `IDE0079	Remove unnecessary suppression`, so can the  `#pragma warning disable xUnit1042` be deleted from `ProjectFileReaderTests.FontConverter.cs`?
image


I tried to delete the `#pragma warning disable xUnit1042`, then compiled, and did not see any warnings about xUnit1042
image

File: src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/System/Windows/Forms/Generators/ProjectFileReaderTests.FontConverter.cs (hunk 2 above)
Comment (LeafShi1) [reply src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/System/Windows/Forms/Generators/ProjectFileReaderTests.FontConverter.cs:60]: Do you mean delete the comment? or resolve the warning here and remove `#pragma warning disable xUnit1042`

File: .editorconfig (hunk 1 above)
Comment (Tanya-Solyanik) [reply .editorconfig:649]: >can the #pragma warning disable xUnit1042 be deleted from ProjectFileReaderTests.FontConverter.cs?

yes, please editorconfig is doing the same thing.

File: src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/System/Windows/Forms/Generators/ProjectFileReaderTests.FontConverter.cs (hunk 2 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/System/Windows/Forms/Generators/ProjectFileReaderTests.FontConverter.cs:60]: Resolve the warning by using typed theory data

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by lonitra — APPROVED** [review]
👍

**Pull request merged**